### PR TITLE
Feature: Base logic for calculation of zki (L1Txs)

### PR DESCRIPTION
**Pull request by Shailu-s**
This PR contains - 

- Updated struct for zki as per discussion.
- Updates for the L1 transaction for calculation of zki.
- Removed the redundant part realted to fee and tokens.

### CODE REVIEW

File: go_sequencer/common/l1tx.go [hunk 1]
```diff
@@ -317,3 +317,87 @@ func L1CoordinatorTxFromBytes(b []byte, chainID *big.Int, tokamakAddress ethComm
...
+	}
+	amountFloat40Bytes, err := amountFloat40.Bytes()
+	if err != nil {
+		return nil, Wrap(err)
+	}
+	copy(b[63:68], amountFloat40Bytes)
+
+	//TODO: Check this with the smart contract team, If we'll get TokenID in the transactions. And if we need to handle that.
+	//Else we can update this for better memory management.
+	// copy(b[68:72], tx.TokenID.Bytes())
```
Comment (Intizar-T) [inline go_sequencer/common/l1tx.go:361]: I'm not sure simply commenting this out would work as expected, especially if this byte array is used for serialization or communication with other systems. We might run into data exchange or processing issues down the line. So my suggestion is to update `RollupConstL1UserTotalBytes` from 78 to 74 length and `copy(b[68:74], toIdxBytes[:])`. But if you think it's safe to simply comment out then we can keep it as it is 😄

File: go_sequencer/common/zk.go [hunk 2]
```diff
@@ -80,9 +81,9 @@ type ZKInputs struct {
...
 	// FromIdx
 	FromIdx []*big.Int `json:"fromIdx"` // uint64 (max nLevels bits), len: [maxTx]
-	// AuxFromIdx is the Idx of the new created account which is
+	// AuxFromAccountIdx is the AccountIdx of the new created account which is
 	// consequence of a L1CreateAccountTx
-	AuxFromIdx []*big.Int `json:"auxFromIdx"` // uint64 (max nLevels bits), len: [maxTx]
+	AuxFromAccountIdx []*big.Int `json:"auxFromIdx"` // uint64 (max nLevels bits), len: [maxTx]
 
 	// ToIdx
 	ToIdx []*big.Int `json:"toIdx"` // uint64 (max nLevels bits), len: [maxTx]
```
Comment (kaiden-araki) [inline go_sequencer/common/zk.go:89]: Here, `FromIdx`, `AuxFromIdx`, and `ToIdx` are all related account, so I think if you want to change `AuxFromIdx` into `AuxFromAccountIdx`, you need to change `FromIdx` and `ToIdx`, too.

File: go_sequencer/common/tx.go [hunk 3]
```diff
@@ -75,25 +75,13 @@ const (
 	// TxTypeExit represents L2->L1 token transfer.  A leaf for this account appears in the exit
 	// tree of the block
 	TxTypeExit TxType = "Exit"
-	// TxTypeTransfer represents L2->L2 token transfer
-	TxTypeTransfer TxType = "Transfer"
```
Comment (Intizar-T) [inline go_sequencer/common/tx.go:79]: Just out of curiosity, does this mean we don't support `Transfer` in the initial version? 🤔

**Comment by Intizar-T** [comment]
Btw I noticed that a lot of the files are red due to some of the tx type removal from `common/tx.go` 😃

File: go_sequencer/common/l1tx.go (hunk 1 above)
Comment (Shailu-s) [reply go_sequencer/common/l1tx.go:361]: Yupp, Thanks @Intizar-T I'll update this.  I left this commented out as it was prior to our discussion, Where i was not sure if we'll keep the tokenID or not.

File: go_sequencer/common/tx.go (hunk 3 above)
Comment (Shailu-s) [reply go_sequencer/common/tx.go:79]: Yeah, @Intizar-T We are not supporting transfer in the initial version

File: go_sequencer/common/zk.go (hunk 2 above)
Comment (Shailu-s) [reply go_sequencer/common/zk.go:89]: Yupp, @kaiden-araki 
The reason of not changing FromIdx and ToIdx was to to leave maximum props as it is and once circom proover is completed then update the same.

File: go_sequencer/common/l1tx.go (hunk 1 above)
Comment (Intizar-T) [reply go_sequencer/common/l1tx.go:353]: @Shailu-s thanks for the detailed answer. Sounds good to me!

**Pull request closed without merging**